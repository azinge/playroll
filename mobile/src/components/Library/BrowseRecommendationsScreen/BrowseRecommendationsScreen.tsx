/**
 * BrowseRecommendationsScreen
 */

import * as React from "react";
import { Text, ActivityIndicator } from "react-native";
import SubScreenContainer from "../../shared/Containers/SubScreenContainer";
import { ListCurrentUserRecommendationsQuery } from "../../../graphql/requests/Recommendation/ListCurrentUserRecommendationsQuery";
import RecommendationCard from "../../shared/Cards/RecommendationCard";
import PlaceholderList from "../../shared/Lists/PlaceholderList";

export default class BrowseRecommendationsScreen extends React.Component {
  render() {
    const extractRecommendations = data => {
      if (
        Object.keys(data).length === 0 ||
        Object.keys(data.private).length === 0
      ) {
        return [];
      }
      return data.private.listCurrentUserRecommendations;
    };
    return (
      <ListCurrentUserRecommendationsQuery>
        {({ loading, error, data }) => {
          const recommendations = extractRecommendations(data);
          return (
            <SubScreenContainer
              title={"My Recommendations"}
              flatList={!loading && !error}
              contentContainerStyle={{ marginTop: 10 }}
              data={recommendations}
              keyExtractor={item => item.id}
              renderItem={({ item }) => {
                return (
                  <RecommendationCard
                    recommendation={item}
                    onPress={() => {}}
                  />
                );
              }}
            >
              {loading && (
                <ActivityIndicator color={"gray"} style={{ paddingTop: 50 }} />
              )}
              {error && (
                <Text style={{ paddingTop: 50 }}>
                  Error Loading Recommendation
                </Text>
              )}
            </SubScreenContainer>
          );
        }}
      </ListCurrentUserRecommendationsQuery>
    );
  }
}
