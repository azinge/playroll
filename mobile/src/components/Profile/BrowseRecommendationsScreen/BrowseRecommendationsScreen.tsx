/**
 * BrowseRecommendationsScreen
 */

import * as React from "react";
import { Text, View, SafeAreaView } from "react-native";

export default class BrowseRecommendationsScreen extends React.Component {
  render() {
    return (
      <SafeAreaView>
        <Text>BrowseRecommendationsScreen</Text>
      </SafeAreaView>
    );
  }
}
