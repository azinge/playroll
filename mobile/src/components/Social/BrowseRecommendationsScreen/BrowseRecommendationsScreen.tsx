/**
 * BrowseRecommendationsScreen
 */

import * as React from 'react';
import { Text, ActivityIndicator, TouchableOpacity } from 'react-native';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';
import SubScreenContainer from '../../shared/Containers/SubScreenContainer';
import { ListCurrentUserRecommendationsQuery } from '../../../graphql/requests/Recommendation/ListCurrentUserRecommendationsQuery';
import RecommendationCard from '../../shared/Cards/RecommendationCard';
import PlaceholderList from '../../shared/Lists/PlaceholderList';
import { Icon } from 'react-native-elements';
import SearchSubHeader from '../../shared/SubHeaders/SearchSubHeader';
import NavigationService from '../../../services/NavigationService';
import EmptyDataFiller from '../../shared/Text/EmptyDataFiller';

export default class BrowseRecommendationsScreen extends React.Component {
  render() {
    const extractRecommendations = data => {
      if (
        !data ||
        Object.keys(data).length === 0 ||
        Object.keys(data.private).length === 0
      ) {
        return [];
      }
      return data.private.listCurrentUserRecommendations;
    };
    return (
      <ListCurrentUserRecommendationsQuery variables={{ offset: 0, count: 20 }}>
        {({ loading, error, data, refetch, fetchMore }) => {
          const recommendations = extractRecommendations(data);
          return (
            <SubScreenContainer
              title={'My Recommendations'}
              flatList
              contentContainerStyle={{
                marginTop: 10,
                paddingBottom: hp('16%'),
              }}
              data={recommendations}
              icons={[
                {
                  name: 'send',
                  onPress: () => {
                    NavigationService.navigate('BrowseSentRecommendations');
                  },
                },
              ]}
              keyExtractor={item => item.id}
              renderFlatListHeader={() => {
                return <SearchSubHeader />;
              }}
              renderFlatListEmptyComponent={() => {
                return loading ? null : (
                  <EmptyDataFiller
                    text={
                      error
                        ? 'Could not load Recommendations'
                        : 'Get your Friends to Recommend you Songs!'
                    }
                    textSize={'h5'}
                    textWidth={300}
                  />
                );
              }}
              renderItem={({ item }) => {
                return (
                  <RecommendationCard
                    recommendation={item}
                    onPress={() => {}}
                  />
                );
              }}
              refreshing={loading}
              onRefresh={() => refetch()}
              onEndReached={() => {
                fetchMore({
                  variables: {
                    offset: recommendations.length,
                  },
                  updateQuery: (prev, { fetchMoreResult }) => {
                    const prevRecommendations = extractRecommendations(prev);
                    const fetchMoreRecommendations = extractRecommendations(
                      fetchMoreResult
                    );
                    if (!fetchMoreResult) return prev;
                    return Object.assign({}, prev, {
                      private: {
                        listCurrentUserRecommendations: [
                          ...prevRecommendations,
                          ...fetchMoreRecommendations,
                        ],
                        __typename: 'PrivateQueryMethods',
                      },
                    });
                  },
                });
              }}
              onEndReachedThreshold={0.5}
            />
          );
        }}
      </ListCurrentUserRecommendationsQuery>
    );
  }
}
