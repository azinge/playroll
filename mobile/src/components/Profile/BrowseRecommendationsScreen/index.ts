import BrowseRecommendationsScreen from "./BrowseRecommendationsScreen";
export default BrowseRecommendationsScreen;
