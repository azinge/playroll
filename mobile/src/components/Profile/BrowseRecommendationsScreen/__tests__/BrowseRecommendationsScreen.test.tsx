import * as React from "react";
import renderer from "react-test-renderer";

import BrowseRecommendationsScreen from "../BrowseRecommendationsScreen";

it("renders correctly", () => {
  const tree = renderer.create(<BrowseRecommendationsScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
