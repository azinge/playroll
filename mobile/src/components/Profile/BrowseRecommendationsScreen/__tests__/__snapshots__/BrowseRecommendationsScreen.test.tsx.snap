// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <Text>
    BrowseRecommendationsScreen
  </Text>
</RCTSafeAreaView>
`;
